tensorbay/opendataset/COVIDChestXRay/loader.py

```diff
@@ -0,0 +1,67 @@
+#!/usr/bin/env python3
+#
+# Copyright 2021 Graviti. Licensed under MIT License.
+#
+# pylint: disable=invalid-name
+# pylint: disable=missing-module-docstring
+
+import csv
+import os
+
+from ...dataset import Data, Dataset
+from ...label import Classification
+
+DATASET_NAME = "COVID-chestxray"
+
+
+def COVIDChestXRay(path: str) -> Dataset:
+    """Dataloader of `COVID-chestxray`_ Dataset.
+
+    .. _COVID-chestxray: https://github.com/ieee8023/covid-chestxray-dataset
+
+    The file structure should be like::
+    <path>
+        images/
+            0a7faa2a.jpg
+            000001-2.png
+            000001-3.jpg
+            1B734A89-A1BF-49A8-A1D3-66FAFA4FAC5D.jpeg
+            ...
+        volumes/
+            coronacases_org_001.nii.gz
+            ....
+        metadata.csv
+        ...
+
+    Arguments:
+        path: The root directory of the dataset.
+
+    Returns:
+        Loaded :class:`~tensorbay.dataset.dataset.Dataset` instance.
+    """
+    root_path = os.path.abspath(os.path.expanduser(path))
+
+    dataset = Dataset(DATASET_NAME)
+    dataset.load_catalog(os.path.join(os.path.dirname(__file__), "catalog.json"))
+    segment = dataset.create_segment()
+
+    csv_path = os.path.join(root_path, "metadata.csv")
+
+    with open(csv_path) as fp:
+        csv_reader = csv.reader(fp)
+        next(csv_reader)
+        for row in csv_reader:
+            folder, image_name = row[22:24]
+            # The 20 images invovled in "volumes" folder currently are invalid to download.
+            if folder == "volumes":
+                continue
+            image_path = os.path.join(root_path, folder, image_name)
+            attribute_content = row[:4] + row[5:22] + row[24:28]
+            attributes = dict(
+                zip(list(dataset.catalog.classification.attributes), attribute_content)
```

Comment:
Why transfer `dataset.catalog.classification.attributes` into a `list`?
Also `attributes` should be a dict, dict is unordered, so that the result of `list` is unstable.